app/routes/dashboard.apps.$appId.security/view.tsx

```diff
@@ -114,6 +124,29 @@ export const SecurityView = ({ endpoint, appId, blockchains }: SecurityViewProps
   const [whitelistMethodsDropdown, setWhitelistMethodsDropdown] = useState<string>("")
   const [whitelistMethodsError, setWhitelistMethodsError] = useState<boolean>(false)
 
+  const [isSecretKeySaveShown, setIsSecretKeySaveShown] = useState<boolean>(false)
```

Comment:
This way is not easily maintainable and scalable (We're falling into useState hell). 
There's a simple and a good solution for this.. take a look here https://www.builder.io/blog/use-reducer

Reply:
good read, thanks @RabeeAbuBaker